nimble/host/mesh/src/proxy.c

```diff
@@ -401,6 +404,21 @@ static void proxy_send_beacons(struct ble_npl_event *work)
 	}
 }
 
+static void proxy_sar_timeout(struct ble_npl_event *work)
+{
+	struct bt_mesh_proxy_client *client;
+	int rc;
+
+	client = ble_npl_event_get_arg(work);
+	assert(client != NULL);
+
+	if ((client->conn_handle != BLE_HS_CONN_HANDLE_NONE)) {
```

Comment:
Wouldn't it be better to cancel the delayed work in the disconnected callback? Otherwise you risk the client object getting reused with another valid handle by the time the timeout expires.

Reply:
Thank you, fixed.